src/Move/Address.h

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "Data.h"
+#include "HexCoding.h"
+#include "PublicKey.h"
+#include <array>
+
+namespace TW::Move {
+template <typename Derived, std::size_t N>
+class Address {
```

Comment:
I don't have a better name but `Move::Address` sounds a little bit weird, other Move chain might not have same address type

Reply:
It's actually named AccountAddress in Rust: https://github.com/move-language/move/blob/61b3aa7c7625968618984e783d820d90115ff694/language/move-core/types/src/account_address.rs#L14 - All Move chains should have the same address type, but could use different feature, aptos use `feature32` and sui `feature20`